frame/system/src/lib.rs

```diff
@@ -515,6 +537,12 @@ decl_module! {
 		}
 
 		/// Set the new changes trie configuration.
+		/// 
+		/// # <weight>
+		/// - `O(L)` where `L` complexity of `deposit_log`
+		/// - 1 storage write or delete (codec `O(1)`).
+		/// - 1 call to `deposit_log`: `O(L)` (which is `O(D)` for the length of `Digest`)
```

Comment:
Maybe better to directly write in the top complexity `O(D)` with D size of Digest. and benchmark will probably take a constant upper size as we don't expect too much digest anyway.

Reply:
This is a tradeoff which I'm unsure how to make. :shrug: 
Because if we refactor `Digest` to support `append` `O(L)` could become `O(1)`.
I guess my solution is sort of "worst of both worlds" because it mentions `O(L)` but also needs to be changed when updating `deposit_log` as well :smile:

Reply:
Went with your variant